src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/RelatedPartyTransactionsQuestionController.java

```diff
@@ -0,0 +1,120 @@
+package uk.gov.companieshouse.web.accounts.controller.smallfull;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import uk.gov.companieshouse.api.ApiClient;
+import uk.gov.companieshouse.api.model.accounts.smallfull.relatedpartytransactions.RelatedPartyTransactionsApi;
+import uk.gov.companieshouse.web.accounts.annotation.NextController;
+import uk.gov.companieshouse.web.accounts.annotation.PreviousController;
+import uk.gov.companieshouse.web.accounts.api.ApiClientService;
+import uk.gov.companieshouse.web.accounts.controller.BaseController;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.smallfull.notes.relatedpartytransactions.RelatedPartyTransactionsQuestion;
+import uk.gov.companieshouse.web.accounts.model.state.CompanyAccountsDataState;
+import uk.gov.companieshouse.web.accounts.service.smallfull.RelatedPartyTransactionsService;
+
+@Controller
+@PreviousController(LoansToDirectorsAdditionalInfoController.class)
+@NextController(OffBalanceSheetArrangementsQuestionController.class)
+@RequestMapping("/company/{companyNumber}/transaction/{transactionId}/company-accounts/{companyAccountsId}/small-full/notes/related-party-transactions-question")
+public class RelatedPartyTransactionsQuestionController extends BaseController {
+
+    private static final String RELATED_PARTY_TRANSACTIONS_QUESTION = "relatedPartyTransactionsQuestion";
+
+    @Autowired
+    private RelatedPartyTransactionsService relatedPartyTransactionsService;
+
+    @Autowired
+    private ApiClientService apiClientService;
+
+    @GetMapping
+    public String getRelatedPartyTransactionsQuestion(Model model,
+                                             @PathVariable String companyNumber,
+                                             @PathVariable String transactionId,
+                                             @PathVariable String companyAccountsId,
+                                             HttpServletRequest request) {
+
+        RelatedPartyTransactionsQuestion relatedPartyTransactionQuestion = new RelatedPartyTransactionsQuestion();
+
+        ApiClient apiClient = apiClientService.getApiClient();
+        try {
+            RelatedPartyTransactionsApi relatedPartyTransactionsApi = relatedPartyTransactionsService.getRelatedPartyTransactions(apiClient, transactionId, companyAccountsId);
```

Comment:
Previously we inflated the _CAS cookie (company data state cookie) to a point where it was too large and was causing issues in live where the cookie would delete an "older" state to make room for a new one; this caused the WillRender method of certain notes to not display and the accounts filing had to be re-started for the user.

Could we try to use a call to the API instead? Or at least discuss whether a call to the API is better or if there could be a way to stop the cookie from getting bloated without causing potential live issues?

Reply:
After a discussion with devs and Divya (lead) we have decided for now to call the API to check if a page should render or if a question has been answered. This will result in minor changes to this PR and eventually a spike/ticket to rework the _CAS cookie.

Reply:
Small update, we will be relying on the _CAS cookie for now for the question page due to the fact we can't have a null selection (no selection made by user yet) if we query the API as it will only be able to return YES/NO....which would mean auto populating NO even if the user hasn't selected anything yet.

This will be investigated in a new ticket surrounding the use of the _CAS cookie.